opencog/util/Config.cc

```diff
@@ -75,334 +75,334 @@ Config::~Config()
+		// fully initialized. Such is life; this is still a lot easier
+		// than debugging the thrown exception in a debugger.
+		logger().warn("No config file found!\n");
+		logger().warn("Searched for \"%s\"\n", search_file().c_str());
+		for (auto& path : search_paths())
+		logger().warn("Searched at %s\n", path.c_str());
+
+		throw IOException(TRACE_INFO,
+			 "unable to open file \"%s\"", filename);
+	}
+
+	_no_config_loaded = false;
+
+	string line;
+	string name;
+	string value;
+	unsigned int line_number = 0;
+	bool have_name = false;
+	bool have_value = false;
+
+	// Read and parse the config file.
+	while (++line_number, fin.good() and getline(fin, line))
+	{
+		string::size_type idx;
+
+		// Find comment and discard the rest of the line.
+		if ((idx = line.find('#')) != string::npos) {
+			line.replace(idx, line.size() - idx, "");
+		}
+
+		// Search for the '=' character.
+		if ((idx = line.find('=')) != string::npos)
+		{
+			// Select name and value
+			name  = line.substr(0, idx);
+			value = line.substr(idx + 1);
+
+			// Strip out whitespace, etc.
+			name  = strip(name);
+			value = strip(value);
+			value = strip(value, "\"");
+
+			have_name = true;
+
+			// Look for trailing comma.
+			if (',' != value[value.size()-1])
+				have_value = true;
+		}
+
+		// Append more to value
+		else if ((string::npos != line.find_first_not_of(blank_chars)) &&
+			have_name &&
+			!have_value)
+		{
+			value += strip(line);
+
+			// Look for trailing comma.
+			if (',' != value[value.size()-1])
+				have_value = true;
+		}
+
+		else if (line.find_first_not_of(blank_chars) != string::npos)
+		{
+			// This might print the diagnostics to the default log file
+			// location.  Note that the config file itself contains the
+			// log file location that is supposed to be used, so this
+			// printing is happening "too early", before the logger is
+			// fully initialized. Such is life; this is still a lot easier
+			// than debugging the thrown exception in a debugger.
+			setup_logger();
+			logger().warn("Invalid config file entry at line %d in %s\n",
+				  line_number, path_where_found().c_str());
+
+			throw InvalidParamException(TRACE_INFO,
+				  "[ERROR] invalid configuration entry (line %d)",
+				  line_number);
+		}
+
+		if (have_name && have_value)
+		{
+			// Finally, store the entries.
+			_table[name] = value;
+			have_name = false;
+			have_value = false;
+			value = "";
+		}
+	}
+	fin.close();
+
+	// Finish configuring the logger... The config file itself
+	// contains the location of the log file. This is working around
+	// a chicken-and-egg problem with reporting config file issues.
+	// Such is life; this is a lot easier than debugging screwed-up
+	// file-path craziness in a debugger. We MUST log the path!!!
+	setup_logger();
+
+	// And then finally, at long last!!! report what happened.
+	logger().info("Using config file found at: %s\n",
+				  path_where_found().c_str());
```

Comment:
Not properly aligned, maybe you haven't used the proper tool/plugin... It looks like the default plugin for smart-tabs in vim is https://www.vim.org/scripts/script.php?script_id=231 (no idea how that works with Neovim).